src/main/java/io/vertx/rabbitmq/impl/RabbitMQClientImpl.java

```diff
@@ -400,6 +404,9 @@ public void stop(Handler<AsyncResult<Void>> resultHandler) {
         log.debug("channel is close, try create Channel");
 
         channel = connection.createChannel();
+
+        if(channelConfirms)
+          channel.confirmSelect();
```

Comment:
I believe that calling again confirmSelect() is useless because you did it already in your confirmSelect method on line 230.